origin-dapp/src/components/navbar.js

```diff
@@ -4,28 +4,60 @@ import { Link } from 'react-router-dom'
 import { FormattedMessage } from 'react-intl'
 
 import { storeWeb3Intent } from 'actions/App'
+import Modal from 'components/modal'
 
 import ConnectivityDropdown from 'components/dropdowns/connectivity'
 import MessagesDropdown from 'components/dropdowns/messages'
 import NotificationsDropdown from 'components/dropdowns/notifications'
 import TransactionsDropdown from 'components/dropdowns/transactions'
 import UserDropdown from 'components/dropdowns/user'
 
+import { clearState } from 'actions/ListingCreate'
+
 class NavBar extends Component {
   constructor(props) {
     super(props)
 
     this.handleLink = this.handleLink.bind(this)
+    this.onAddNew = this.onAddNew.bind(this)
+    this.onContinue = this.onContinue.bind(this)
+    this.showDraftModal = this.showDraftModal.bind(this)
     this.state = {
       noWeb3Account: false,
       notWeb3EnabledDesktop: false,
-      notWeb3EnabledMobile: false
+      notWeb3EnabledMobile: false,
+      showDraftModal: false
     }
   }
 
   handleLink(e) {
     this.props.storeWeb3Intent('create a listing')
+    if (!web3.givenProvider || !this.props.web3Account) {
+      e.preventDefault()
+    }
+  }
+
+  showDraftModal() {
+    this.setState({
+      showDraftModal: true
+    })
+  }
```

Comment:
I'm not seeing the draft modal pop up when I have a draft in localstorage. When I click "Add a Listing", it takes me straight to the listing create review page